deploy/snapshot/cmd/cuda-vmm-interpose/coordinator.c

```
@@ -0,0 +1,680 @@
+      fgets(line, sizeof(line), input) == NULL || sscanf(line, "job %32s", job) != 1 || !valid_id(job))
+    return -1;
+  while (fgets(line, sizeof(line), input) != NULL) {
+    struct participant* participant;
+    char id[DYN_VMM_ID_SIZE];
+    unsigned int record_count;
+    unsigned int index;
+
+    if (sscanf(line, "participant %32s %u", id, &record_count) != 2 || !valid_id(id) ||
+        record_count > DYN_VMM_MAX_RECORDS)
+      goto failed;
+    participants = realloc(participants, (count + 1) * sizeof(*participants));
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)